doc/manual/contracts.md

```diff
@@ -1058,3 +1057,148 @@ is wrapping the original value with delayed checks inside**. Doing so preserves
 laziness of the language and only triggers the checks when the values are used
 or exported in a configuration. This is the reason for general custom contracts
 to have an immediate part and a delayed part.
+
+## Boolean combinators
+
+The stdlib features a set of boolean combinators for contracts:
+`std.contract.all_of`, `std.contract.any_of` and `std.contract.not`.
+
+It's important to note that those combinators are only *approximations* of what
+you could expect from boolean operations. Fortunately, they are overstrict
+approximation, in that they might reject more values, but will never let invalid
+values slip through.
+
+The reason is, once again, the presence of delayed contracts.
+
+### Combining delayed contracts
+
+Let `Foo` be the contract `std.contract.any_of [{ foo | String }, {foo |
+Number}]`. `any_of` can't evaluate the field `foo` to determine if it's a number
```

Comment:
I was going to object to the word "can't" here, because in principle `any_of` could use `deep_seq` to force the evaluation. But that seems to only allow for applying the inner contract in the "blaming" way, not in the "checking" way. Maybe it would be useful to have a mechanism for forcing inner contract checks? I can imagine that sometimes you might really want the exact semantics but you don't care about laziness.

Edit: ok, I read to the end and maybe the future strict contracts make this unnecessary

Reply:
I think the "can't" here is meant as "assuming that this contract must preserve lazyness". Of course, it can otherwise extract the `foo` field and `apply_as_custom String` to get an immediate answer (as does the example of a custom contract that is more eager later, I believe?). But it might be good to emphasize it, I'll repeat the assumption of lazyness.

Reply:
Improved in 78b877c23e4fdb7334459e0591f6964d98c1cb05